src/main/java/hw4/pages/MetalAndColorPage.java

```diff
@@ -0,0 +1,109 @@
+package hw4.pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+public class MetalAndColorPage extends AbstractPage {
+
+    public MetalAndColorPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(css = "p.radio > label")
+    private List<WebElement> summaryRadioButtons;
+
+    @FindBy(css = "p.checkbox > label")
+    private List<WebElement> elementsCheckboxes;
+
+    @FindBy(css = "button[title='Colors']")
+    private WebElement colorDropdown;
+
+    @FindBy(xpath = "//button[@title='Metals']//span[2]")
+    private WebElement metalDropdown;
+
+    @FindBy(xpath = "//ul[@class='dropdown-menu inner selectpicker']//li")
+    private List<WebElement> colorAndMetalOptions;
+
+    @FindBy(xpath = "//button[@class='btn btn-default dropdown-toggle']")
+    private WebElement vegetablesDropdown;
+
+    @FindBy(css = ".dropdown-menu > li > a > label")
+    private List<WebElement> vegetableCheckboxes;
+
+    @FindBy(css = "#submit-button")
+    private WebElement submitButton;
+
+    @FindBy(xpath = "//ul[@class='panel-body-list results']//li")
+    private List<WebElement> results;
+
+    public void selectSummaryRadioButtons(List<String> summaryRadioButtonName) {
+        for (WebElement element : summaryRadioButtons) {
+            if (summaryRadioButtonName.contains(element.getText())) {
+                element.click();
+            }
+        }
+    }
+
+    public void selectElementsCheckbox(List<String> elementsName) {
+        for (WebElement element : elementsCheckboxes) {
+            if (elementsName.contains(element.getText())) {
+                element.click();
+            }
+        }
+    }
+
+    public void clickColorDropdown() {
+        colorDropdown.click();
+    }
+
+    public void selectColorDropdown(String colorText) {
+        for (WebElement element : colorAndMetalOptions) {
+            if (element.getText().equals(colorText)) {
+                element.click();
+            }
+        }
+    }
+
+    public void clickMetalDropdown() {
+        metalDropdown.click();
+    }
```

Comment:
Why do you do not use `Select` element?

Reply:
The type of this element is a button, there is no opportunity to select an option directly